### ISSUE TITLE
Scribble is not compatible with the @openzepplin/hardhat-upgrades plugin

### ISSUE THREAD

**Issue by cd1m0** [ConsenSys/scribble#202]
On repos with the [`@openzeppelin/hardhat-upgrades`](https://www.npmjs.com/package/@openzeppelin/hardhat-upgrades) plugin, people run into the following error:

```
__scribble_ReentrancyUtils.sol:11: Variable `__scribble_out_of_contract` is assigned an initial value
    Move the assignment to the initializer
```

It looks like our utils conract is using a non-upgradeability-friendly pattern in the constructor:

```
    bool __scribble_out_of_contract = true;
```

Apparently upgradeable contracts don't allow initialization of state variables anywhere outside the `initialize()` method.

The simplest solution so far seems to be to change `__scribble_out_of_contract` to `__scribble_in_contract`, so that the variable doesn't need initialization. (i.e. the default value - `false` - is a correct initial value).

**Comment by 0xmoostorm** [comment]
Additionally, also ran into the following error: 
```
@openzeppelin/contracts-upgradeable/__scribble_ReentrancyUtils.sol:11: Inserted `__scribble_out_of_contract`
  > New variables should be placed after all existing inherited variables
  ```

A possible solution would be to append the `__scribble_out_of_contract` to the target contract that is being instrumented? Is it truly necessary to generate  `__scribble_ReentrancyUtils.sol`, can the events and variable be added to the target contract instead?

**Comment by cd1m0** [comment]
We need the `__scribble_out_of_contract` variable to be shared by all base and child contracts in an inheritance chain. The simplest way we came up to do this was to place the variable in a base contract inherited by all contracts in the hierarchy. Implementing it differently is a lot trickier...

**Comment by cd1m0** [comment]
Another way to try and potentially fix this is instead of having a solidity-level variable for marking whether we are in a contract, would be to select a storage address and use assembly `SSTORE`/`SLOAD` to that address to store the boolean marker. We can either pick an address at random and hope it doesn't clash with anything, or analyze the storage layouts of all available contracts and pick some address that is likely out of the way. Of course we can never account for aliasing a map access. Not sure how good this solution is.

**Comment by cd1m0** [comment]
Should be fixed by #210
